coderd/workspaceagents.go

```diff
@@ -1723,10 +1724,9 @@ func ellipse(v string, n int) string {
 // @Security CoderSessionToken
 // @Accept json
 // @Tags Agents
-// @Param request body agentsdk.PostMetadataRequest true "Workspace agent metadata request"
-// @Param key path string true "metadata key" format(string)
+// @Param request body []agentsdk.PostMetadataRequest true "Workspace agent metadata request"
 // @Success 204 "Success"
-// @Router /workspaceagents/me/metadata/{key} [post]
+// @Router /workspaceagents/me/metadata [post]
 // @x-apidocgen {"skip": true}
```

Comment:
I think we need to keep the old endpoint around for legacy agents for at least one release. Older agents will continue to post to `/metadata/{key}` and get a 404.